setup-test-namespace.sh

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env sh
+
+NAMESPACE=${1:=micronaut-kubernetes}
+DELETE_NAMESPACE=${2:=false}
+
+echo "Configuration:"
+echo "\tNAMESPACE:        ${NAMESPACE}"
+echo "\tDELETE_NAMESPACE: ${DELETE_NAMESPACE}"
+
+#
+# Delete namespace if exists and it is requested by second parameter
+# that is either "true" or "yes"
+if [ "${DELETE_NAMESPACE}" = "true" ] || [ "${DELETE_NAMESPACE}" = "yes" ]; then
+  kubectl get ns ${NAMESPACE} > /dev/null 2>&1
+  if [ $? -eq 0 ]; then
+    echo "Deleting namespace ${NAMESPACE}"
+    kubectl delete ns ${NAMESPACE} --wait=true
+  fi
+fi
+
+# Create namespace and switch context to that namespace
+kubectl create namespace ${NAMESPACE}
+kubectl config set-context --current --namespace=${NAMESPACE}
+
+# Create role and role binding
+kubectl create role service-discoverer --verb=get,list,watch --resource=services,endpoints,configmaps,secrets,pods
+kubectl create rolebinding default-service-discoverer --role service-discoverer --serviceaccount=${NAMESPACE}:default
+
+# Create secrets and config maps
+kubectl create configmap game-config-properties --from-file=kubernetes-discovery-client/src/test/resources/k8s/game.properties
+kubectl create configmap game-config-yml --from-file=kubernetes-discovery-client/src/test/resources/k8s/game.yml
+kubectl create configmap game-config-json --from-file=kubernetes-discovery-client/src/test/resources/k8s/game.json
+kubectl create configmap literal-config --from-literal=special.how=very --from-literal=special.type=charm
+kubectl create secret generic test-secret --from-literal=username='my-app' --from-literal=password='39528$vdg7Jb'
+kubectl create secret generic another-secret --from-literal=secretProperty='secretValue'
+kubectl create secret generic mounted-secret --from-literal=mountedVolumeKey='mountedVolumeValue'
+
+kubectl label configmap game-config-yml app=game
+kubectl label configmap literal-config app=game
+kubectl label secret another-secret app=game
+
+# Create and expose deployment - example service
+cat <<EOF | kubectl apply -f -
```

Comment:
Could these YAML snippets be stored on their own YAML files?

Reply:
ok